src/Shared/UnitTests/MockEngine.cs

```diff
@@ -399,8 +413,7 @@ internal void AssertLogContains(string contains)
         {
             if (_upperLog == null)
             {
-                _upperLog = _log;
-                _upperLog = _upperLog.ToUpperInvariant();
+                _upperLog = _log.ToString().ToUpperInvariant();
```

Comment:
I guess it's fine to write and forget about _upperLog because the engine is only used once? If the same engine is passed to multiple tasks, then _upperLog can get stale